{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!/usr/local/cluster/bin/python3\n",
    "\n",
    "import sys, re\n",
    "from io import StringIO\n",
    "import pandas as pd\n",
    "import csv\n",
    "from Bio import SeqIO\n",
    "from BCBio import GFF\n",
    "\n",
    "#take in arguments from command line, argument 1 is the significant SNP list, \n",
    "#argument 2 is the GFF file, argument 3 is the candidate gene list to write out\n",
    "f1 = sys.argv[1]\n",
    "f2 = sys.argv[2]\n",
    "f3 = sys.argv[3]\n",
    "\n",
    "#write output to file named in arguments\n",
    "output = open(f3, \"w\")\n",
    "\n",
    "gff_handle = open(f2,\"r\")\n",
    "gff_iter = gff_handle.readlines()\n",
    "with open(f1) as csvfile:\n",
    "        sig_snps_reader = csv.reader(csvfile, delimiter=',')\n",
    "        next(sig_snps_reader)\n",
    "        for row in sig_snps_reader:\n",
    "                SNP = row[2]\n",
    "                chrom = row[3].strip()\n",
    "                base = row[4].strip()\n",
    "                fbase = float(base)\n",
    "                fchrom = float(chrom)\n",
    "                search_pos_start = fbase - 7000\n",
    "                search_pos_end = fbase + 7000\n",
    "                if fchrom == 1:\n",
    "                        gffchrom = 'NC_012870.2'\n",
    "                        print(gffchrom)\n",
    "                if fchrom == 2:\n",
    "                        gffchrom = 'NC_012871.2'\n",
    "                        print(gffchrom)\n",
    "                if fchrom == 3:\n",
    "                        gffchrom = 'NC_012872.2'\n",
    "                        print(gffchrom)\n",
    "                if fchrom == 4:\n",
    "                        gffchrom = 'NC_012873.2'\n",
    "                        print(gffchrom)\n",
    "                if fchrom == 5:\n",
    "                        gffchrom = 'NC_012874.2'\n",
    "                        print(gffchrom)\n",
    "                if fchrom == 6:\n",
    "                        gffchrom = 'NC_012875.2'\n",
    "                        print(gffchrom)\n",
    "                if fchrom == 7:\n",
    "                        gffchrom = 'NC_012876.2'\n",
    "                        print(gffchrom)\n",
    "                if fchrom == 8:\n",
    "                        gffchrom = 'NC_012877.2'\n",
    "                        print(gffchrom)\n",
    "                if fchrom == 9:\n",
    "                        gffchrom = 'NC_012878.2'\n",
    "                        print(gffchrom)\n",
    "                if fchrom == 10:\n",
    "                        gffchrom = 'NC_012879.2'\n",
    "                        print(gffchrom)\n",
    "                for line in gff_iter:\n",
    "                        if gffchrom in line:\n",
    "                                parse_line = line.split()\n",
    "                                if '##sequence-region' not in parse_line:\n",
    "                                        feat_start = float(parse_line[3])\n",
    "                                        feat_stop = float(parse_line[4])\n",
    "                                        if feat_start in range(int(search_pos_start), int(search_pos_end)):\n",
    "                                            distance = feat_start - fbase\n",
    "                                                distance_str = str(distance)\n",
    "                                                output.write(SNP + '\\t' + chrom + '\\t' + base + '\\t' + row[12] + '\\t' + parse_line[0] + '\\t' + parse_line[1] + '\\t' + parse_line[2] + '\\t' + parse_line[3] + '\\t' + parse_line[4] + '\\t' + parse_line[6] + '\\t' + distance_str + '\\t' + parse_line[8] + '\\n')\n",
    "                                elif feat_stop in range(int(search_pos_start), int(search_pos_end)):\n",
    "                                                distance = feat_stop - fbase\n",
    "                                                distance_str = str(distance)\n",
    "                                                output.write(SNP + '\\t' + chrom + '\\t' + base + '\\t' + row[12] + parse_line[0] + '\\t' + parse_line[1] + '\\t' + parse_line[2] + '\\t' + parse_line[3] + '\\t' + parse_line[4] + '\\t' + parse_line[6] + '\\t' + distance_str + '\\t' + parse_line[8] +  '\\n')\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
